store/root/store.go

```diff
@@ -162,7 +161,7 @@ func (s *Store) Query(storeKey string, version uint64, key []byte, prove bool) (
 		defer s.telemetry.MeasureSince(now, "root_store", "query")
 	}
 
-	val, err := s.stateStore.Get(storeKey, version, key)
+	val, err := s.stateStorage.Get(storeKey, version, key)
```

Comment:
Didn't we discuss in Slack to add a `s.stateStorage == nil`  check?

Reply:
we discussed keep stateStorage not `nil`? I also updated the test-cases to ensure not nil

Reply:
I thought we wanted to add it for safety.